src/java/org/apache/cassandra/index/sai/disk/v2/sortedbytes/SortedBytesReader.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.v2.sortedbytes;
+
+import java.io.IOException;
+import java.util.Iterator;
+
+import org.apache.cassandra.io.util.FileHandle;
+import org.apache.cassandra.utils.Pair;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.cassandra.utils.bytecomparable.ByteSource;
+import org.apache.lucene.store.ByteArrayIndexInput;
+import org.apache.lucene.store.IndexInput;
+import org.apache.lucene.store.RandomAccessInput;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.LongValues;
+import org.apache.lucene.util.packed.DirectMonotonicReader;
+
+import static org.apache.cassandra.index.sai.disk.v2.sortedbytes.SortedBytesWriter.DIRECT_MONOTONIC_BLOCK_SHIFT;
+import static org.apache.cassandra.index.sai.disk.v2.sortedbytes.SortedBytesWriter.TERMS_DICT_BLOCK_MASK;
+import static org.apache.cassandra.index.sai.disk.v2.sortedbytes.SortedBytesWriter.TERMS_DICT_BLOCK_SHIFT;
+
+/**
+ * Lookup a point id by ByteComparable.
+ *   Implemented exactly from Lucene version 7.5 SortedDocValues
+ *
+ * Lookup a ByteComparable by point id.
+ *   Implemented using the trie with a long payload for the point id.
+ */
+public class SortedBytesReader
+{
+    private final SortedBytesWriter.Meta meta;
+    private final FileHandle trieHandle;
+    private final BytesRef term;
```

Comment:
Are there any purposes of this field other than pure optimization - that is avoiding heap allocations?